STMicroelectronics/STM32L4_L4+/app/sprinkler/src/sprinkler_v1i1_interface.c

```diff
@@ -0,0 +1,158 @@
+
+/*
+ * Concrete implementations of the sprinkler 1 water_now commands.
+ */
+static az_result sprinkler_1_water_now_concrete(az_ulib_model_in model_in, az_ulib_model_out model_out)
+{
+  (void)model_out;
+  const sprinkler_1_water_now_model_in* const in = (const sprinkler_1_water_now_model_in* const)model_in;
+  return sprinkler_v1i1_water_now(in->timer);
+}
+
+/*
+ * Concrete implementations of the sprinkler 1 stop commands.
+ */
+static az_result sprinkler_1_stop_concrete(az_ulib_model_in model_in, az_ulib_model_out model_out)
+{
+  (void)model_in;
+  (void)model_out;
+  return sprinkler_v1i1_stop();
+}
+
+static az_result sprinkler_1_water_now_span_wrapper(az_span model_in_span, az_span* model_out_span)
+{
+  az_result result = AZ_OK;
+
+  AZ_ULIB_TRY
+  {
+    // Unmarshalling JSON to model_in_span.
+    az_json_reader jr;
+    sprinkler_1_water_now_model_in water_now_model_in = { 0 };
+    AZ_ULIB_THROW_IF_AZ_ERROR(az_json_reader_init(&jr, model_in_span, NULL));
+    AZ_ULIB_THROW_IF_AZ_ERROR(az_json_reader_next_token(&jr));
+    while (jr.token.kind != AZ_JSON_TOKEN_END_OBJECT)
+    {
+      if (az_json_token_is_text_equal(&jr.token, AZ_SPAN_FROM_STR(SPRINKLER_1_TIMER_NAME)))
+      {
+        AZ_ULIB_THROW_IF_AZ_ERROR(az_json_reader_next_token(&jr));
+        AZ_ULIB_THROW_IF_AZ_ERROR(az_span_atoi32(jr.token.slice, &(water_now_model_in.timer)));
+      }
+      AZ_ULIB_THROW_IF_AZ_ERROR(az_json_reader_next_token(&jr));
+    }
+    AZ_ULIB_THROW_IF_AZ_ERROR(AZ_ULIB_TRY_RESULT);
+
+    // Call.
+    result = sprinkler_1_water_now_concrete((az_ulib_model_in)&water_now_model_in, NULL);
+
+    // Marshalling model_out to JSON.
+    *model_out_span = az_span_create_from_str("{}");
+  }
+  AZ_ULIB_CATCH(...) { result = AZ_ULIB_TRY_RESULT; }
+
+  return result;
+}
+
+static az_result sprinkler_1_stop_span_wrapper(az_span model_in_span, az_span* model_out_span)
+{
+  az_result result = AZ_OK;
+
+  AZ_ULIB_TRY
+  {
+    // Unmarshalling JSON to model_in.
+    (void)model_in_span;
+
+    // Call.
+    result = sprinkler_1_stop_concrete(NULL, NULL);
+
+    // Marshalling model_out to JSON.
+    *model_out_span = az_span_create_from_str("{}");
+  }
+  AZ_ULIB_CATCH(...) { result = AZ_ULIB_TRY_RESULT; }
+
+  return result;
+}
+
+static const az_ulib_capability_descriptor SPRINKLER_1_CAPABILITIES[SPRINKLER_1_CAPABILITY_SIZE] = {
+  AZ_ULIB_DESCRIPTOR_ADD_COMMAND(
+      SPRINKLER_1_WATER_NOW_COMMAND_NAME,
+      sprinkler_1_water_now_concrete),
+  AZ_ULIB_DESCRIPTOR_ADD_COMMAND(SPRINKLER_1_STOP_COMMAND_NAME, sprinkler_1_stop_concrete)
+};
+
+static const az_ulib_interface_descriptor SPRINKLER_1_DESCRIPTOR = AZ_ULIB_DESCRIPTOR_CREATE(
+    SPRINKLER_1_INTERFACE_NAME,
+    SPRINKLER_1_INTERFACE_VERSION,
+    SPRINKLER_1_CAPABILITY_SIZE,
+    sprinkler_1_call_w_str,
+    SPRINKLER_1_CAPABILITIES);
+
+static az_result sprinkler_1_call_w_str(az_span name, az_span model_in_span, az_span* model_out_span)
```

Comment:
>sprinkler_1_call_w_str [](start = 17, length = 22)

All functions xxx_call_w_str will be replaced by a code in the IPC that will call directly the xxx_span_wrapper functions.